test/list/reduce.wren

```diff
@@ -0,0 +1,9 @@
+var a = [1, 4, 2, 1, 5]
+var max = new Fn {|a, b| a > b ? a : b }
+var sum = new Fn {|a, b| a + b }
```

Comment:
>It would be good to have a test that demonstrates the order that the fold happens.

I wasn't aware right-to-left folds are a thing. In Haskell, both fold variants fold left-to-right (as this is the only sensible thing to do with lists):
```haskell
Prelude> foldl (++) "" ["W", "o", "r", "l", "d"]
"World"
Prelude> foldr (++) "" ["W", "o", "r", "l", "d"]
"World"
```
in contrast to JS:
```javascript
["W", "o", "r", "l", "d"].reduce(function(a, b) { return a + b })
"World"
["W", "o", "r", "l", "d"].reduceRight(function(a, b) { return a + b })
"dlroW"
```
I guess right-associative folds aren't useful outside of lazy languages.